File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/main/DocsFavoritesFragment.java
package app.editors.manager.ui.fragments.main;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import app.editors.manager.managers.providers.CloudFileProvider;

public class DocsFavoritesFragment extends DocsCloudFragment {

    public static DocsFavoritesFragment newInstance() { return new DocsFavoritesFragment(); }

    public static final String ID = CloudFileProvider.Section.Favorites.getPath();

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init();
    }

    @Override
    protected boolean onSwipeRefresh() {
        if (!super.onSwipeRefresh()) {
            mCloudPresenter.getItemsById(ID);
            return true;
        }

        return false;
    }

    @Override
    public void onScrollPage() {
        super.onScrollPage();
        if (mCloudPresenter.getStack() == null) {
            mCloudPresenter.getItemsById(ID);
        }
    }

    @Override
    public void onStateEmptyBackStack() {
        super.onStateEmptyBackStack();
        if (mSwipeRefresh != null) {
            mSwipeRefresh.setRefreshing(true);
        }
        mCloudPresenter.getItemsById(ID);
    }
    @Override
    public void onRemoveItemFromFavorites() {
        mCloudPresenter.removeFromFavorites();
    }

    private void init() {
        mCloudPresenter.checkBackStack();
    }
}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/main/DocsFavoritesFragment.kt
package app.editors.manager.ui.fragments.main

import android.os.Bundle
import android.view.View
import app.documents.core.network.ApiContract
import app.editors.manager.managers.providers.CloudFileProvider

class DocsFavoritesFragment : DocsCloudFragment() {

    companion object {
        val ID = CloudFileProvider.Section.Favorites.path

        fun newInstance(account: String): DocsFavoritesFragment {
            return DocsFavoritesFragment().apply {
                arguments = Bundle(1).apply {
                    putString(KEY_ACCOUNT, account)
                }
            }
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    override fun onSwipeRefresh(): Boolean {
        if (!super.onSwipeRefresh()) {
            cloudPresenter.getItemsById(ID)
            return true
        }
        return false
    }

    override fun onScrollPage() {
        super.onScrollPage()
        if (cloudPresenter.stack == null) {
            cloudPresenter.getItemsById(ID)
        }
    }

    override val section: Int
        get() = ApiContract.SectionType.UNKNOWN

    override fun onStateEmptyBackStack() {
        super.onStateEmptyBackStack()
        swipeRefreshLayout?.isRefreshing = true
        cloudPresenter.getItemsById(ID)
    }

    override fun onUpdateFavoriteItem() {
        cloudPresenter.removeFromFavorites()
    }

    private fun init() {
        cloudPresenter.checkBackStack()
    }
}